### ISSUE TITLE
Unexpected game freeze

### ISSUE THREAD

**Issue by secretluverr** [CleverRaven/Cataclysm-DDA#66670]
### Describe the bug

Game freezes when traveling to a certain area on the map.  htop shows one CPU at 100% usage while frozen so I suspect it's an unintended infinite loop.  

debug.log shows: 

```
17:08:08.158 : Starting log.
17:08:08.158 INFO : Cataclysm DDA version 0.G
17:08:08.159 INFO : [main] C locale set to en_US.UTF-8
17:08:08.159 INFO : [main] C++ locale set to en_US.UTF-8
17:08:08.159 INFO : SDL version used during compile is 2.26.3
17:08:08.159 INFO : SDL version used during linking and in runtime is 2.28.1
17:08:08.194 INFO : Number of render drivers on your system: 3
17:08:08.194 INFO : Render driver: 0/opengl
17:08:08.194 INFO : Render driver: 1/opengles2
17:08:08.194 INFO : Render driver: 2/software
17:08:08.195 INFO : [options] C locale set to en_US.UTF-8
17:08:08.195 INFO : [options] C++ locale set to en_US.UTF-8
17:08:08.195 INFO : Setting language to: 'en'
17:08:08.195 INFO : [i18n] Scanning core translations from /usr/share/locale
17:08:08.195 INFO : [i18n] Scanning mod translations from /home/secretluver/.local/share/cataclysm-dda/mods/
17:08:08.195 WARNING : opendir [/home/secretluver/.local/share/cataclysm-dda/mods/] failed with "No such file or directory".
17:08:08.214 WARNING : opendir [/usr/share/cataclysm-dda/cache] failed with "No such file or directory".
17:08:08.377 INFO : Active renderer: 2/software
17:08:08.713 INFO : USE_COLOR_MODULATED_TEXTURES is set to 0
17:08:08.735 INFO : Setting language to: 'en'
17:08:08.750 WARNING : opendir [/home/secretluver/.local/share/cataclysm-dda/mods] failed with "No such file or directory".
17:08:12.535 WARNING : opendir [/home/secretluver/.local/share/cataclysm-dda/save/Grand Isle/mods] failed with "No such file or directory".
17:08:13.036 INFO : src/mod_tracker.h:68: Tried check if 'FakeSpecial_s_gas' had a duplicate, but type 'overmap_special' does not track object sources
17:08:18.012 INFO : Loaded tileset: Chibi_Ultica
17:08:18.130 INFO : Loaded tileset: Chibi_Ultica
17:10:26.344 ERROR : (error message will follow backtrace)
    cataclysm-tiles(debug_write_backtrace(std::ostream&)+0x3e) [0x562a9ab5ddcf]
    cataclysm-tiles(DebugLog(DebugLevel, DebugClass)+0x23a) [0x562a9ab6335a]
    cataclysm-tiles(realDebugmsg(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x17a) [0x562a9ab65145]
    cataclysm-tiles(+0x5b8a4c) [0x562a9a905a4c]
    cataclysm-tiles(debug_menu::debug()+0x2bff) [0x562a9ab8f1dc]
    cataclysm-tiles(game::do_regular_action(action_id&, avatar&, cata::optional<tripoint> const&)+0x3d4e) [0x562a9acc7c52]
    cataclysm-tiles(game::handle_action()+0xc16) [0x562a9acc9998]
    cataclysm-tiles(do_turn()+0x6ef) [0x562a9abadab1]
    cataclysm-tiles(main+0x1b38) [0x562a9a5cf777]
    /usr/lib/libc.so.6(+0x23850) [0x7fb522039850]
    /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7fb52203990a]
    cataclysm-tiles(_start+0x25) [0x562a9a8fb4a5]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    ??:?
    DebugLog(DebugLevel, DebugClass)
    ??:?
    realDebugmsg(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    ??:?
    achievements_tracker::~achievements_tracker()
    ??:?
    debug_menu::debug()
    ??:?
    game::do_regular_action(action_id&, avatar&, cata::optional<tripoint> const&)
    ??:?
    game::handle_action()
    ??:?
    do_turn()
    ??:?
    main
    ??:?
    ??
    ??:0
    ??
    ??:0
    _start
    ??:?
Backtrace emission took 0 seconds.
(continued from above) ERROR : src/debug_menu.cpp:3130 [void debug_menu::debug()] Test debugmsg
```

### Attach save file

[Grand Isle-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11952547/Grand.Isle-trimmed.tar.gz)

### Steps to reproduce

1. Kill all creatures (Not necessary, just easier for debugging)
2. Travel south to nearest road
3. Travel left (West) on map.
4. Freeze should happen before you make it to 4-way intersection.

### Expected behavior

Game should not freeze.  

### Screenshots

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/8518143/b27cc445-a121-4422-8cc9-729474a3b37c)

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: 0.G [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Bionic Slots [cbm_slots],
    SpeedyDex [speedydex],
    Tamable Wildlife [Tamable_Wildlife],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

### Additional context

crash.log doesn't show up for this instance.  Debug log added to description.

**Comment by SurFlurer** [comment]
Fixed by #64996 . That should become a backport candidate.

**Comment by RenechCDDA** [comment]
Duplicate of #64799

**Comment by RenechCDDA** [comment]
Specifically in your save it's this task:

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/b24b0cf1-9f69-4c89-a109-c2ca696c1443)

**Comment by secretluverr** [comment]
@RenechCDDA @SurFlurer Thank you
